unity-renderer/Assets/DCLPlugins/ECS7/Systems/PointerInputSystem/ECSPointerInputSystem.cs

```diff
@@ -56,242 +47,178 @@ private enum InputHitType
-                                                   || state.lastInputHover.entity.entityId != colliderData.entity.entityId
-                                                   || state.lastInputHover.sceneNumber != colliderData.scene.sceneData.sceneNumber;
-
-                        // was other entity previously hovered?
-                        if (isPreviouslyHoveredEntity && isHoveringNewEntity)
-                        {
-                            bool isValidScene = colliderData.scene.sceneData.sceneNumber == state.lastInputHover.sceneNumber
-                                                || state.worldState.ContainsScene(state.lastInputHover.sceneNumber);
-
-                            if (isValidScene)
-                            {
-                                AddInputResultEvent(
-                                    state,
-                                    currentPointerInput,
-                                    state.lastInputHover.scene,
-                                    state.lastInputHover.entity.entityId,
-                                    raycastRay,
-                                    raycastHit,
-                                    PointerEventType.PetHoverLeave
-                                );
-                            }
-                        }
-
-                        // hover enter
-                        if (isHoveringNewEntity)
-                        {
-                            state.lastInputHover.entity = colliderData.entity;
-                            state.lastInputHover.scene = colliderData.scene;
-                            state.lastInputHover.sceneNumber = colliderData.scene.sceneData.sceneNumber;
-                            state.lastInputHover.hasValue = true;
-
-                            AddInputResultEvent(
-                                state,
-                                currentPointerInput,
-                                colliderData.scene,
-                                colliderData.entity.entityId,
-                                raycastRay,
-                                raycastHit,
-                                PointerEventType.PetHoverEnter
-                            );
-                        }
-                        break;
+                    }
+
+                    BroadcastInputResultEvent(
+                        state,
+                        inputAction,
+                        -1,
+                        raycastRay,
+                        raycastHit,
+                        pointerEventType,
+                        colliderScene
+                    );
+
+                    // update
+                    prevState[i] = curState[i];
                 }
             }
-            else // no entity hit
+
+            // Check if the hovered entity has changed with three options:
+
+            // 1) We were hitting a collider A and now we're hitting a collider B
```

Comment:
Could we get those steps extracted into methods? so we can reduce the visual complexity of this whole Update function.

Reply:
### Example
```C#
if (IsColliderDifferent(state))
{
    HandleColliderChanged(state);
} 
if (IsColliderMissing(state))
{ 
    HandleMissingCollider(state);
}
if (IsColliderAvailable(state))
{
    HandleAvailableCollider(state)
} 
```